components/eventing-controller/pkg/backend/utils/utils_v2.go

```diff
@@ -0,0 +1,79 @@
+//nolint:gosec
+package utils
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	eventingv1alpha2 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+
+	"go.uber.org/zap"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/pkg/errors"
+
+	apigatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+)
+
+type EventTypeInfo struct {
+	OriginalType  string
+	CleanType     string
+	ProcessedType string
+}
+
+// NameMapper is used to map Kyma-specific resource names to their corresponding name on other
+// (external) systems, e.g. on different eventing backends, the same Kyma subscription name
+// could map to a different name.
+type NameMapper interface {
+	MapSubscriptionName(subscriptionName, subscriptionNamespace string) string
+}
+
+func getExposedURLFromAPIRule(apiRule *apigatewayv1beta1.APIRule, targetURL string) (string, error) {
```

Comment:
(nit) I still think this is very eventmesh specific and can go into the eventmesh utils. I think it is here right now since it is used by beb as well and to avoid cyclic imports/clashes. Maybe a TODO to move this to the eventmash utils?